org/postgresql/ssl/jdbc4/LibPQFactory.java

```diff
@@ -281,9 +283,45 @@ public boolean verify(String hostname, SSLSession session) {
         } else {
           return false;
         }
-      } else {
-        return CN.equals(hostname);
+      } else if(CN.equals(hostname.toLowerCase())){
+        return true;
       }
+      try {
+        for(List<?> i: serverCert.getSubjectAlternativeNames())
+        {
+          int type = (Integer) i.get(0);
+            switch(type)
+            {
+              case 2:
+                String dNSName = (String) i.get(1);
+                if(dNSName.toLowerCase().equals(hostname.toLowerCase()))
```

Comment:
Just adding my thoughts... 

Old school scholars always thought that while comparing it's always good to have toLowerCase / toUpperCase before compare. 

Now due to JVM performance improvements everything is changed.

- `equalsIgnoreCase`  is more readable
- If one ore two letters having different case then there will not be major performance boost. See [JMH perf test](http://stackoverflow.com/questions/29785600/efficiency-of-equalsignorecase-versus-touppercase-equals-and-tolowercase-e) (Although example is relatively simple and having only one letter having different case)